src/tcp/Compression.jl

```diff
@@ -0,0 +1,54 @@
+using CodecLz4
+
+@enum Compression::UInt8 begin
+    COMPRESSION_NONE = 0
+    COMPRESSION_CHECKSUM_ONLY = 0x02
+    COMPRESSION_LZ4 = 0x82
+end
+
+Compression(flag::Bool)::Compression = flag ? COMPRESSION_LZ4 : COMPRESSION_NONE
+
+function Compression(name::String)::Compression
+    if lowercase(name) == "lz4"
+        return COMPRESSION_LZ4
+    elseif lowercase(name) == "checksum_only"
+        return COMPRESSION_CHECKSUM_ONLY
+    end
+    error("unkown compression mode: $(name)")
+end
+
+"""compress data according to the compression mode"""
+function compress(mode::Compression, data)::Vector{UInt8}
```

Comment:
Should we specify the type of `data` here?

Reply:
yes, not sure why I dropped it.